nx/lib/nx.ex

```diff
@@ -7016,10 +7054,15 @@ defmodule Nx do
       end
 
     t =
-      if axis do
-        sort(tensor, axis: axis)
-      else
-        tensor |> flatten() |> sort()
+      case {shape, axis} do
+        {{}, _} ->
+          reshape(tensor, {1})
```

Comment:
Doesn't this mean that this PR changes the way default flatten works?

I wouldn't expect having to change anywhere thag calls Nx.flatten for adding the new feature

Reply:
It does, but I think the original behavior was wrong. I would not expect `Nx.flatten` on a scalar to create a new dimension